src/NServiceBus.Core/DelayedDelivery/TimeoutManager/DispatchTimeoutBehavior.cs

```diff
@@ -16,12 +15,11 @@ public DispatchTimeoutBehavior(IDispatchMessages dispatcher, IPersistTimeouts pe
             dispatchConsistency = GetDispatchConsistency(transportTransactionMode);
         }
 
-        protected override async Task Terminate(ISatelliteProcessingContext context)
+        public async Task Invoke(PushContext context)
         {
-            var message = context.Message;
-            var timeoutId = message.Headers["Timeout.Id"];
+            var timeoutId = context.Headers["Timeout.Id"];
 
-            var timeoutData = await persister.Peek(timeoutId, context.Extensions).ConfigureAwait(false);
+            var timeoutData = await persister.Peek(timeoutId, context.Context).ConfigureAwait(false);
```

Comment:
That would be a breaking change in the seam. I've added it to the POA of https://github.com/Particular/V6Launch/issues/69 to make sure we deal with it when we add onError